data/strings/sound.txt

```diff
@@ -114,6 +115,21 @@
     zh-Hans = 向右急转。
     zh-Hant = 向右急轉。
 
+  [make_a_right_turn_street]
```

Comment:
Does it make sense to embed this translation into dist_direction_onto_street instead of defining it separately?

Reply:
No in fact we need to make a bunch of other _street strings, probably one for every possible direction. In at least Japanese and Dutch we have a situation where a command like turn left or take the 3rd exit has different grammar from a preposition like turn left onto Main Street, it'll end up being more like "left onto Main Street make a turn." So we need separate strings for that and only that case.

Reply:
If I'm really lucky, "make a turn" and "take an exit" won't be additional complications requiring even more assembled particles. Will have to double check with the JA and NL experts. But yeah duplicating every direction for every language seems excessive.

Reply:
Also, think about easier support and translation to other languages. Sometimes it is easier to write several similar strings than to understand what these special characters mean and how your algorithm combines them together.

Reply:
Absolutely, if it gets any more complex or if people struggle then I'll do it. I just don't want to give dozens of people dozens of things to duplicate if it can easily be avoided. Placeholders and fragments in i18n are pretty common.

Reply:
How many almost identical strings will be now if you remove one placeholder?

Reply:
My proposed solution is that we only duplicate directional strings (with a _street suffix) for those languages who have different syntax for commands versus prepositional phrases, like Japanese and Dutch. 90% of languages will need no such duplication so there will only be a few dozen new strings:

```
  [make_a_slight_right_turn]
    en = Keep right.
  [make_a_slight_right_turn_street]
    en = keep right

  [make_a_right_turn]
    en = Turn right.
  [make_a_right_turn_street]
    en = turn right

  [make_a_sharp_right_turn]
    en = Sharp right.
  [make_a_sharp_right_turn_street]
    en = sharp right

  [enter_the_roundabout]
    en = Enter the roundabout.
  [enter_the_roundabout_street]
    en = enter the roundabout

//...

  [destination]
    en = You’ll arrive.
  [destination_street]
    en = you’ll arrive

  [dist_direction_onto_street]
    en = %1$s %2$s onto %3$s
```

to do your proposed method of removing the %2 directional placeholder, we'd remove the new dist_direction_onto_street translation but need to duplicate every directional string in every language; the sounds.txt file will be 2x larger if not 3x or 4x to handle some future cases like the desired "exit 123 for city center toward main st" alternative to the current "exit onto 123, city center, main street":

```
  [make_a_slight_right_turn]
    en = Keep right.
  [in_dist_make_a_slight_right_turn_street]
    en = %1$s keep right onto %3$s

  [make_a_right_turn]
    en = Turn right.
  [in_dist_make_a_right_turn_street]
    en = %1$s turn right onto %3$s

  [make_a_sharp_right_turn]
    en = Sharp right.
  [in_dist_make_a_sharp_right_turn_street]
    en = %1$s sharp right onto %3$s

  [enter_the_roundabout]
    en = Enter the roundabout.
  [in_dist_enter_the_roundabout_street]
    en = %1$s enter the roundabout onto %3$s

//...

  [destination]
    en = You’ll arrive.
  [in_dist_destination_street]
    en = %1$s you’ll arrive onto %3$s
```

I think the best solution is the one currently coded. It's not that uncommon to have multiple placeholders as long as we communicate with our translators properly, it's not too hard to program for, and it opens up a lot of excellent future language possibility.

Reply:
Ok, let's polish, cleanup, and test it properly then. Very good comments and examples for these translations are needed in the sounds.txt for other translators too.